tests/integration/security/util/cert/cert.go

```diff
@@ -0,0 +1,136 @@
+//  Copyright 2020 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cert
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/shell"
+	"istio.io/istio/tests/integration/security/util/dir"
+	"istio.io/istio/tests/util"
+	"istio.io/pkg/log"
+)
+
+// GetCertOutput gets the certificate output from openssl s-client command.
+func GetCertOutput(ns namespace.Instance, fromSelector, fromContainer, connectTarget string) (string, error) {
+	retry := util.Retrier{
+		BaseDelay: 10 * time.Second,
+		Retries:   3,
+		MaxDelay:  30 * time.Second,
+	}
+
+	fromPod, err := dir.GetPodName(ns, fromSelector)
+	if err != nil {
+		return "", fmt.Errorf("err getting the from pod name: %v", err)
+	}
+
+	var out string
+	retryFn := func(_ context.Context, i int) error {
+		execCmd := fmt.Sprintf(
+			"kubectl exec %s -c %s -n %s -- openssl s_client -showcerts -connect %s",
```

Comment:
Do we need to do this? Does our docs do this? This will now allow us to switch to using distroless images

Reply:
Yes, we need to do this and the reason to do this step is to obtain the certificate chain on the sidecar provisioned through SDS. The test verifies that the root certificate is the example CA certificate plugged in. Our document https://istio.io/docs/tasks/security/plugin-ca-cert/#verifying-the-certificates also does this step.